src/main/java/com/example/videotecha/model/RegisteredUser.java

```diff
@@ -0,0 +1,127 @@
+package com.example.videotecha.model;
+
+import com.example.videotecha.model.enums.UserRole;
+import jakarta.persistence.*;
+
+import java.util.Objects;
+
+@Entity
+public class RegisteredUser {
+
+    @Id
+    @SequenceGenerator(name = "userSequenceGenerator", sequenceName = "userSequenceGenerator", initialValue = 1, allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "userSequenceGenerator")
+    @Column(name="reg_user_id", unique=true, nullable=false)
+    private Long id;
+
+    @Column(nullable=false)
+    private String firstName;
+
+    @Column(nullable=false)
+    private String lastName;
+
+    @Column(unique=true, nullable=false)
+    private String email;
+
+    @Column(unique=true, nullable=false)
+    private String username;
+
+    @Column(nullable=false)
+    private String password;
+
+    @Column(nullable=false)
+    private UserRole role;
```

Comment:
This can be `@Enumerated(value = EnumType.STRING)` for a stringified db representation